bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/BalRuntime.java

```diff
@@ -48,9 +62,68 @@
 public class BalRuntime extends Runtime {
 
     private final Scheduler scheduler;
+    private final Module module;
+    private boolean moduleInitialized = false;
 
-    public BalRuntime(Scheduler scheduler) {
+    public BalRuntime(Scheduler scheduler, Module module) {
         this.scheduler = scheduler;
+        this.module = module;
+    }
+
+    public BalRuntime(Module module) {
+        this.scheduler = new Scheduler(false);
+        this.module = module;
+    }
+
+    public void init() {
+        // Invoke Config init
+        String configClassName = getConfigClassName(module.getOrg(), module.getName(), module.getMajorVersion());
+        Class<?> configClazz;
+        try {
+            configClazz = Class.forName(configClassName);
+        } catch (Throwable e) {
+            throw ErrorCreator.createError(StringUtils.fromString("failed to load configuration class :" +
+                    configClassName));
+        }
+        TomlDetails configDetails = LaunchUtils.getConfigurationDetails();
+        Object response = directInvoke(configClazz,
+                new Class[]{String[].class, Path[].class, String.class},
+                new Object[]{new String[]{}, configDetails.paths, configDetails.configContent});
+        if (response instanceof Throwable) {
+            throw ErrorCreator.createError(StringUtils.fromString("configurable initialization failed due to " +
+                    RuntimeUtils.formatErrorMessage((Throwable) response)), (Throwable) response);
+        }
+        // Invoke module init
+        invokeMethodAsync("$moduleInit", new Object[1], null, PredefinedTypes.TYPE_NULL, "init");
+        moduleInitialized = true;
+    }
+
+    public void start() {
+        if (!moduleInitialized) {
+            throw ErrorCreator.createError(StringUtils.fromString("failed to complete module initialization"));
+        }
+        invokeMethodAsync("$moduleStart", new Object[1], null, PredefinedTypes.TYPE_NULL, "start");
+    }
+
+    public void invokeMethodAsync(String functionName, Object[] args, Callback callback) {
```

Comment:
We may ignore handling these cases for all three cases (start, function invocation and stop) as well since it'll throw a Ballerina runtime error `value creator object not found for the module X`. WDYT @warunalakshitha?

Reply:
As we discussed offline, we decided to have the error in the above cases as the value creator errors can be misleading.